src/basic/list.h

```diff
@@ -169,3 +169,18 @@
 
 #define LIST_IS_EMPTY(head)                                             \
         (!(head))
+
+/* Join two lists tail to head: a->b, c->d to a->b->c->d and de-initialise second list */
+#define LIST_JOIN(name,a,b)                                             \
+        do {                                                            \
+                assert(b);                                              \
+                if (!a)                                                 \
```

Comment:
always enclose every use of parameters in `()` to avoid issues with operator priorities and such. just as a matter of style in macros

Reply:
Thanks, fixed